crates/proof-of-sql/src/proof_primitive/dory/dynamic_dory_structure.rs

```diff
@@ -30,22 +30,62 @@
 //! ...
 //! ```
 
+/// Sets the data to indicate if no index exists.
+#[allow(dead_code)]
+const fn no_index() -> usize {
+    usize::MAX
+}
+
+/// Returns if the index exists.
+#[allow(dead_code)]
+pub(crate) const fn index_exists(index: usize) -> bool {
+    index != no_index()
+}
+
+/// Returns the full width of a row in the matrix.
```

Comment:
Can you add docs for the behavior for row #0 and row#1, since those are unusual?

Reply:
I added a note, similar to the one added to the `row_start_index` function. I am assuming that is the extent of docs to be added? Do we want this documented elsewhere?